tap_linkedin_ads/client.py

```diff
@@ -195,8 +220,32 @@ def fetch_and_set_access_token(self):
 
         data = response.json()
         self.__access_token = data['access_token']
+
+        # data['expires_in'] is an integer of seconds until the access_token expires.
+        # Technically this self.__expires is inaccurate because it was true when LinkedIn generated the token, but
+        # we receive and process that response some (very) small amount of time after it was true.
         self.__expires = datetime.utcnow() + timedelta(seconds=data['expires_in'])
-        LOGGER.info('Authorized, token expires = %s', format(self.__expires))
+
+
+    def fetch_and_set_access_token(self):
+
+        # If refresh token is not provided then we are assuming that it is an old connection
+        # and client has provided the valid access_token already
+        if not self.__refresh_token:
+            return
+
+        if self.__access_token:
+
+            if not self.__expires:
```

Comment:
Rather than checking __expires exists and populating it if it doesn't, you could rig it up such that fetch_and_set_token_expires checks if __expires is populated, populates it if needed, and returns it's value. That way, if you need to check if the token you have is expired, you can just call that (rather than having to check if you have the expiration every time you want to use __expires).